packages/core/services/HttpServiceBase/index.ts

```diff
@@ -210,6 +210,30 @@ export default class HttpServiceBase {
         return new RestServiceResponse(response.data);
     }
 
+    public async put<T>(url: string, body: string): Promise<RestServiceResponse<T>> {
+        const encodedUrl = HttpServiceBase.encodeURI(url);
+        const config = { headers: { "Content-Type": "application/json" } };
+
+        let response;
+        try {
+            // if this fails, bubble up exception
+            response = await retry.execute(() => this.httpClient.put(encodedUrl, body, config));
+        } catch (err) {
+            // Specific errors about the failure from services will be in this path
+            if (axios.isAxiosError(err) && err?.response?.data?.message) {
+                throw new Error(JSON.stringify(err.response.data.message));
+            }
+            throw err;
+        }
+
+        if (response.status >= 400 || response.data === undefined) {
```

Comment:
Could you have a valid response with no data?

Reply:
Another case of using the pattern but not necessarily being able to defend. I believe we expect to always receive at least the state of the data from a success response